File: app/src/main/java/com/kota/ASFramework/PageController/ASViewController.java
package com.kota.ASFramework.PageController;

import android.content.Context;
import android.content.Intent;
import android.content.res.AssetManager;
import android.content.res.Resources;
import android.view.View;
import java.util.Iterator;
import java.util.Vector;

/* loaded from: classes.dex */
public abstract class ASViewController {
  public static final int ALERT_LEFT_BUTTON = -1;
  public static final int ALERT_MIDDLE_BUTTON = -3;
  public static final int ALERT_RIGHT_BUTTON = -2;
  private ASNavigationController _controller = null;
  protected boolean _is_loaded = false;
  private ASPageView _page_view = null;
  private boolean _contains_in_container = false;
  private boolean _marked_removed = false;
  private boolean _marked_added = false;
  private boolean _is_appeared = false;
  private boolean _is_disappeared = true;
  private boolean _is_request_refresh = false;
  private Vector<ASViewControllerOperationListener> _operation_listeners = null;
  private Vector<ASViewControllerAppearListener> _appear_listeners = null;
  private Vector<ASViewControllerDisappearListener> _disappear_listeners = null;

  public abstract int getPageLayout();

  public void setPageView(ASPageView aPageView) {
    if (this._page_view != null) {
      this._page_view.setOwnerController(null);
    }
    this._page_view = aPageView;
    if (this._page_view != null) {
      this._page_view.setOwnerController(this);
    }
  }

  public ASPageView getPageView() {
    return this._page_view;
  }

  public int getPageType() {
    return 0;
  }

  public void onPageDidLoad() {
  }

  public void onPageWillAppear() {
  }

  public void onPageDidAppear() {
  }

  public void onPageRefresh() {
  }

  public void onPageFreeMemory() {
  }

  public void onPageWillDisappear() {
  }

  public void onPageDidDisappear() {
  }

  public void onPageDidRemoveFromNavigationController() {
  }

  public void onPageDidAddToNavigationController() {
  }

  public void onPageDidUnload() {
  }

  public void notifyPageDidRemoveFromNavigationController() {
    if (this._operation_listeners != null) {
      Vector<ASViewControllerOperationListener> listeners = new Vector<>(this._operation_listeners);
      Iterator<ASViewControllerOperationListener> it = listeners.iterator();
      while (it.hasNext()) {
        ASViewControllerOperationListener listener = it.next();
        if (listener != null) {
          listener.onASViewControllerWillRemoveFromNavigationController(this);
        }
      }
    }
    onPageDidRemoveFromNavigationController();
  }

  public void notifyPageDidAddToNavigationController() {
    if (this._operation_listeners != null) {
      Vector<ASViewControllerOperationListener> listeners = new Vector<>(this._operation_listeners);
      Iterator<ASViewControllerOperationListener> it = listeners.iterator();
      while (it.hasNext()) {
        ASViewControllerOperationListener listener = it.next();
        if (listener != null) {
          listener.onASViewControllerWillAddToNavigationController(this);
        }
      }
    }
    onPageDidAddToNavigationController();
  }

  public void notifyPageWillAppear() {
    this._is_disappeared = false;
    if (this._appear_listeners != null) {
      Vector<ASViewControllerAppearListener> listeners = new Vector<>(this._appear_listeners);
      Iterator<ASViewControllerAppearListener> it = listeners.iterator();
      while (it.hasNext()) {
        ASViewControllerAppearListener listener = it.next();
        if (listener != null) {
          listener.onASViewControllerWillAppear(this);
        }
      }
    }
    onPageWillAppear();
  }

  public void notifyPageDidAppear() {
    this._is_appeared = true;
    if (this._appear_listeners != null) {
      Vector<ASViewControllerAppearListener> listeners = new Vector<>(this._appear_listeners);
      Iterator<ASViewControllerAppearListener> it = listeners.iterator();
      while (it.hasNext()) {
        ASViewControllerAppearListener listener = it.next();
        if (listener != null) {
          listener.onASViewControllerDidAppear(this);
        }
      }
    }
    onPageDidAppear();
    if (this._is_request_refresh) {
      onPageRefresh();
      this._is_request_refresh = false;
    }
  }

  public void notifyPageWillDisappear() {
    this._is_appeared = false;
    if (this._disappear_listeners != null) {
      Vector<ASViewControllerDisappearListener> listeners = new Vector<>(this._disappear_listeners);
      Iterator<ASViewControllerDisappearListener> it = listeners.iterator();
      while (it.hasNext()) {
        ASViewControllerDisappearListener listener = it.next();
        if (listener != null) {
          listener.onASViewControllerWillDisappear(this);
        }
      }
    }
    onPageWillDisappear();
  }

  public void notifyPageDidDisappear() {
    this._is_disappeared = true;
    if (this._disappear_listeners != null) {
      Vector<ASViewControllerDisappearListener> listeners = new Vector<>(this._disappear_listeners);
      Iterator<ASViewControllerDisappearListener> it = listeners.iterator();
      while (it.hasNext()) {
        ASViewControllerDisappearListener listener = it.next();
        if (listener != null) {
          listener.onASViewControllerDidDisappear(this);
        }
      }
    }
    onPageDidDisappear();
  }

  public void requestPageRefresh() {
    if (this._is_appeared) {
      onPageRefresh();
    } else {
      this._is_request_refresh = true;
    }
  }

  public boolean isPageAppeared() {
    return this._is_appeared;
  }

  public boolean isPageDisappeared() {
    return this._is_disappeared;
  }

  public boolean backToMainToolbarAfterExtendToolbarClicked() {
    return true;
  }

  protected void finalize() throws Throwable {
    onPageDidUnload();
    super.finalize();
  }

  public void clear() {
  }

  public Context getContext() {
    return this._controller;
  }

  public Resources getResource() {
    return this._controller.getResources();
  }

  public ASNavigationController getNavigationController() {
    return this._controller;
  }

  public View findViewById(int viewID) {
    if (this._page_view != null) {
      return this._page_view.findViewById(viewID);
    }
    return null;
  }

  public Object getSystemService(String name) {
    if (this._controller != null) {
      return this._controller.getSystemService(name);
    }
    return null;
  }

  /* JADX INFO: Access modifiers changed from: protected */
  public void onSizeChanged(int newWidth, int newHeight, int oldWidth, int oldHeight) {
  }

  /* JADX INFO: Access modifiers changed from: protected */
  public boolean onBackPressed() {
    getNavigationController().popViewController();
    return true;
  }

  /* JADX INFO: Access modifiers changed from: protected */
  public boolean onSearchButtonClicked() {
    return false;
  }

  /* JADX INFO: Access modifiers changed from: protected */
  public boolean onMenuButtonClicked() {
    return false;
  }

  public void startActivity(Intent intent) {
    if (this._controller != null) {
      this._controller.startActivity(intent);
    }
  }

  public void setNavigationController(ASNavigationController controller) {
    this._controller = controller;
  }

  public void reloadLayout() {
    if (this._controller != null) {
      this._controller.reloadLayout();
    }
  }

  public boolean onReceivedGestureUp() {
    return false;
  }

  public boolean onReceivedGestureDown() {
    return false;
  }

  public boolean onReceivedGestureLeft() {
    return false;
  }

  public boolean onReceivedGestureRight() {
    return false;
  }

  public boolean isTopPage() {
    return getNavigationController() != null && getNavigationController().getTopController() == this;
  }

  public AssetManager getAssets() {
    return this._controller.getAssets();
  }

  public boolean containsInContainer() {
    return this._contains_in_container;
  }

  public void setContainsInContainer(boolean contains) {
    this._contains_in_container = contains;
  }

  /* JADX INFO: Access modifiers changed from: protected */
  public boolean isMarkedRemoved() {
    return this._marked_removed;
  }

  /* JADX INFO: Access modifiers changed from: protected */
  public boolean isMarkedAdded() {
    return this._marked_added;
  }

  /* JADX INFO: Access modifiers changed from: protected */
  public void prepareForRemove() {
    this._marked_removed = true;
  }

  /* JADX INFO: Access modifiers changed from: protected */
  public void prepareForAdd() {
    this._marked_removed = false;
    this._marked_added = true;
  }

  /* JADX INFO: Access modifiers changed from: protected */
  public void cleanMark() {
    this._marked_removed = false;
    this._marked_added = false;
  }

  public String toString() {
    return "ASViewController[Type:" + getPageType() + "]";
  }

  public void registerPageOperationListener(ASViewControllerOperationListener aListener) {
    if (this._operation_listeners == null) {
      this._operation_listeners = new Vector<>();
    }
    this._operation_listeners.add(aListener);
  }

  public void unregisterPageOperationListener(ASViewControllerOperationListener aListener) {
    if (this._operation_listeners != null) {
      this._operation_listeners.remove(aListener);
    }
  }

  public void registerAppearListener(ASViewControllerAppearListener aListener) {
    if (this._appear_listeners == null) {
      this._appear_listeners = new Vector<>();
    }
    this._appear_listeners.add(aListener);
  }

  public void unregisterAppearListener(ASViewControllerAppearListener aListener) {
    if (this._appear_listeners != null) {
      this._appear_listeners.remove(aListener);
    }
  }

  public void registerDisappearListener(ASViewControllerDisappearListener aListener) {
    if (this._disappear_listeners == null) {
      this._disappear_listeners = new Vector<>();
    }
    this._disappear_listeners.add(aListener);
  }

  public void unregisterDisappearListener(ASViewControllerDisappearListener aListener) {
    if (this._disappear_listeners != null) {
      this._disappear_listeners.remove(aListener);
    }
  }
}

File: app/src/main/java/com/kota/asFramework/pageController/ASViewController.kt
package com.kota.asFramework.pageController

import android.content.Intent
import android.content.res.AssetManager
import android.content.res.Resources
import android.view.View
import java.util.Vector

abstract class ASViewController {
    lateinit var navigationController: ASNavigationController
    protected var isLoaded: Boolean = false
    private var aSPageView: ASPageView? = null
    private var isContainedInContainer = false
    var isMarkedRemoved: Boolean = false
        private set
    var isMarkedAdded: Boolean = false
        private set
    var isPageAppeared: Boolean = false
        private set
    var isPageDisappeared: Boolean = true
        private set
    private var isRequestRefresh = false
    private var operationListeners: Vector<ASViewControllerOperationListener?>? = null
    private var appearListeners: Vector<ASViewControllerAppearListener?>? = null
    private var disappearListeners: Vector<ASViewControllerDisappearListener?>? = null

    abstract val pageLayout: Int

    var pageView: ASPageView?
        get() = this.aSPageView
        set(aPageView) {
            if (this.aSPageView != null) {
                this.aSPageView!!.ownerController = null
            }
            this.aSPageView = aPageView
            if (this.aSPageView != null) {
                this.aSPageView!!.ownerController = this
            }
        }

    open val pageType: Int
        get() = 0

    open fun onPageDidLoad() {
    }

    open fun onPageWillAppear() {
    }

    open fun onPageDidAppear() {
    }

    open fun onPageRefresh() {
    }

    fun onPageFreeMemory() {
    }

    open fun onPageWillDisappear() {
    }

    open fun onPageDidDisappear() {
    }

    open fun onPageDidRemoveFromNavigationController() {
    }

    fun onPageDidAddToNavigationController() {
    }

    open fun onPageDidUnload() {
    }

    fun notifyPageDidRemoveFromNavigationController() {
        if (this.operationListeners != null) {
            val listeners = Vector<ASViewControllerOperationListener?>(this.operationListeners!!)
            for (listener in listeners) {
                listener?.onASViewControllerWillRemoveFromNavigationController(this)
            }
        }
        onPageDidRemoveFromNavigationController()
    }

    fun notifyPageDidAddToNavigationController() {
        if (this.operationListeners != null) {
            val listeners = Vector<ASViewControllerOperationListener?>(this.operationListeners!!)
            for (listener in listeners) {
                listener?.onASViewControllerWillAddToNavigationController(this)
            }
        }
        onPageDidAddToNavigationController()
    }

    fun notifyPageWillAppear() {
        this.isPageDisappeared = false
        if (this.appearListeners != null) {
            val listeners = Vector<ASViewControllerAppearListener?>(this.appearListeners!!)
            for (listener in listeners) {
                listener?.onASViewControllerWillAppear(this)
            }
        }
        onPageWillAppear()
    }

    fun notifyPageDidAppear() {
        this.isPageAppeared = true
        if (this.appearListeners != null) {
            val listeners = Vector<ASViewControllerAppearListener?>(this.appearListeners!!)
            for (listener in listeners) {
                listener?.onASViewControllerDidAppear(this)
            }
        }
        onPageDidAppear()
        if (this.isRequestRefresh) {
            onPageRefresh()
            this.isRequestRefresh = false
        }
    }

    fun notifyPageWillDisappear() {
        this.isPageAppeared = false
        if (this.disappearListeners != null) {
            val listeners = Vector<ASViewControllerDisappearListener?>(this.disappearListeners!!)
            for (listener in listeners) {
                listener?.onASViewControllerWillDisappear(this)
            }
        }
        onPageWillDisappear()
    }

    fun notifyPageDidDisappear() {
        this.isPageDisappeared = true
        if (this.disappearListeners != null) {
            val listeners = Vector<ASViewControllerDisappearListener?>(this.disappearListeners!!)
            for (listener in listeners) {
                listener?.onASViewControllerDidDisappear(this)
            }
        }
        onPageDidDisappear()
    }

    fun requestPageRefresh() {
        if (this.isPageAppeared) {
            onPageRefresh()
        } else {
            this.isRequestRefresh = true
        }
    }

    fun backToMainToolbarAfterExtendToolbarClicked(): Boolean {
        return true
    }

    @Throws(Throwable::class)
    protected fun finalize() {
        onPageDidUnload()
    }

    open fun clear() {
    }

    val context: ASNavigationController?
        get() = this.navigationController

    val resource: Resources?
        get() = this.navigationController.resources

    fun findViewById(viewID: Int): View? {
        if (this.aSPageView != null) {
            return this.aSPageView!!.findViewById(viewID)
        }
        return null
    }

    fun getSystemService(name: String): Any? {
        return this.navigationController.getSystemService(name)
    }

    fun onSizeChanged(newWidth: Int, newHeight: Int, oldWidth: Int, oldHeight: Int) {
    }

    open fun onBackPressed(): Boolean {
        this.navigationController.popViewController()
        return true
    }

    open fun onSearchButtonClicked(): Boolean {
        return false
    }

    open fun onMenuButtonClicked(): Boolean {
        return false
    }

    fun startActivity(intent: Intent?) {
        this.navigationController.startActivity(intent)
    }

    fun startActivityForResult(intent: Intent?, code: Int) {
        this.navigationController.startActivityForResult(intent, code)
    }

    fun reloadLayout() {
        this.navigationController.reloadLayout()
    }

    fun onReceivedGestureUp(): Boolean {
        return false
    }

    fun onReceivedGestureDown(): Boolean {
        return false
    }

    fun onReceivedGestureLeft(): Boolean {
        return false
    }

    open fun onReceivedGestureRight(): Boolean {
        return false
    }

    open val isTopPage: Boolean
        get() = this.navigationController.topController === this

    val assets: AssetManager?
        get() = this.navigationController.assets

    fun containsInContainer(): Boolean {
        return this.isContainedInContainer
    }

    fun setContainsInContainer(contains: Boolean) {
        this.isContainedInContainer = contains
    }

    fun prepareForRemove() {
        this.isMarkedRemoved = true
    }

    fun prepareForAdd() {
        this.isMarkedRemoved = false
        this.isMarkedAdded = true
    }

    fun cleanMark() {
        this.isMarkedRemoved = false
        this.isMarkedAdded = false
    }

    override fun toString(): String {
        return "ASViewController[Type:" + this.pageType + "]"
    }

    fun registerPageOperationListener(aListener: ASViewControllerOperationListener) {
        if (this.operationListeners == null) {
            this.operationListeners = Vector<ASViewControllerOperationListener?>()
        }
        this.operationListeners!!.add(aListener)
    }

    fun unregisterPageOperationListener(aListener: ASViewControllerOperationListener) {
        if (this.operationListeners != null) {
            this.operationListeners!!.remove(aListener)
        }
    }

    fun registerAppearListener(aListener: ASViewControllerAppearListener) {
        if (this.appearListeners == null) {
            this.appearListeners = Vector<ASViewControllerAppearListener?>()
        }
        this.appearListeners!!.add(aListener)
    }

    fun unregisterAppearListener(aListener: ASViewControllerAppearListener) {
        if (this.appearListeners != null) {
            this.appearListeners!!.remove(aListener)
        }
    }

    fun registerDisappearListener(aListener: ASViewControllerDisappearListener?) {
        if (this.disappearListeners == null) {
            this.disappearListeners = Vector<ASViewControllerDisappearListener?>()
        }
        this.disappearListeners!!.add(aListener)
    }

    fun unregisterDisappearListener(aListener: ASViewControllerDisappearListener?) {
        if (this.disappearListeners != null) {
            this.disappearListeners!!.remove(aListener)
        }
    }

    companion object {
        const val ALERT_LEFT_BUTTON: Int = -1
        const val ALERT_MIDDLE_BUTTON: Int = -3
        const val ALERT_RIGHT_BUTTON: Int = -2
    }
}